File: coding_test/avatar/chess.py
n=input()
steps=[(2,1),(2,-1),(-2,-1),(-2,-1),(1,-2),(-1,2),(1,2),(-1,-2)]
row = ord(n[0])-97+1

column = int(n[1])
cnt = 0
for x,y in steps:
    if(1<=row+x<=8 and 1<=column+y<=8):
        cnt += 1

print(cnt)                                                                                                                                                    

File: coding_test/avatar/chess.cpp
#include <iostream>
#include <string>
using namespace std;

int main()
{
    int result = 0;
    int dx[] = {-2, -2, 2, 2, -1, 1, -1, 1};
    int dy[] = {-1, 1, -1, 1, -2, -2, 2, 2};
    string w;
    cin >> w;
    int row = w[0] - 'a' + 1;
    int column = w[1] - '0';
    for (int i = 0; i < 8; i++)
    {
        int x = row + dx[i];
        int y = column + dy[i];
        if (x >= 1 && x <= 8 && y >= 1 && y <= 8)
        {
            result += 1;
        }
    }
    cout << result;
}